### PR TITLE
Change internal command line exception classes to package-protected RuntimeExceptions

### PR DESCRIPTION
**Pull request by cmnbroad**
and change their base class to RuntimeException so that all CommandLineExceptions can be treated as user errors.

### CODE REVIEW

File: src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java [hunk 1]
```diff
@@ -76,8 +76,9 @@ private static String getMessage(final double minValue, final double maxValue, f
      * Class CommandLineParserInternalException
      * <p/>
      * For internal errors in the command line parser not related to syntax errors in the command line itself.
+     * Package protected to restrict usage to the arg parsers.
      */
-    public static class CommandLineParserInternalException extends CommandLineException {
+    static class CommandLineParserInternalException extends RuntimeException {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java]: Should probably be public to allow clients to catch it.

File: src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java [hunk 2]
```diff
@@ -90,16 +91,17 @@ public CommandLineParserInternalException( final String s, final Throwable throw
 
     /**
      * For wrapping errors that are believed to never be reachable
+     * Package protected to restrict usage to the arg parsers.
      */
-    public static class ShouldNeverReachHereException extends CommandLineException {
+    static class ShouldNeverReachHereException extends RuntimeException {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java]: Should probably extend `CommandLineParserInternalException`, which should be the superclass of all internal errors.

File: src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java (hunk 1 above)
Comment (cmnbroad) [inline src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java]: Done.

File: src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java (hunk 2 above)
Comment (cmnbroad) [inline src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java]: Done.

**Comment by droazen** [comment]
:+1: merge when ready

**Pull request merged**